subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/SkipCachedTaskExecuter.java

```diff
@@ -81,7 +77,7 @@ public void execute(final TaskInternal task, TaskStateInternal state, TaskExecut
                 if (taskState.isAllowedToUseCachedResults()) {
                     try {
                         OriginTaskExecutionMetadata originMetadata = buildCache.load(
-                            buildCacheCommandFactory.createLoad(cacheKey, outputProperties, task, taskProperties, taskOutputsGenerationListener, taskState, clock)
+                            buildCacheCommandFactory.createLoad(cacheKey, outputProperties, task, taskProperties, taskOutputsGenerationListener, taskState, context.getExecutionTimer())
```

Comment:
Why change the timer here? The only effect is that we report a different time on the info log (time it took from start of task to finish loading from cache vs time it took to load and unpack the result).